File: AkkaRouterExample/src/main/java/org/akka/essentials/router/example/RandomTimeActor.java
package org.akka.essentials.router.example;

import java.util.Random;
import java.util.concurrent.TimeUnit;

import akka.actor.UntypedActor;

public class RandomTimeActor extends UntypedActor {

	Random randomGenerator = new Random();

	@Override
	public void onReceive(Object msg) throws Exception {
		int sleepTime = randomGenerator.nextInt(5);
		System.out.println(String.format("Actor # '%s' sleeping for '%s'",
				getSelf().path().name(), sleepTime));
		TimeUnit.SECONDS.sleep(sleepTime);
		getSender().tell("Message from Actor #" + getSelf().path());
	}

}

File: AkkaRouterExample/src/main/scala/org/akka/essentials/router/example/RandomTimeActor.scala
package org.akka.essentials.router.example
import akka.actor.Actor
import java.util.Random
import java.util.concurrent.TimeUnit
import akka.actor.ActorLogging

class RandomTimeActor extends Actor with ActorLogging{
  val randomGenerator = new Random()
  def receive: Receive = {
    case message =>
      val sleepTime = randomGenerator.nextInt(5)
      log.info("Actor # {} will return in {}", self.path.name, sleepTime)
      TimeUnit.SECONDS.sleep(sleepTime);
      sender.tell("Message from Actor #" + self.path)
  }
}